### PR TITLE
config and reduce Tabulation space

### PR DESCRIPTION
**Pull request by nashousni**

### CODE REVIEW

File: gse-afs-ext-base/src/main/resources/lang/ModificationScript_fr.properties [hunk 1]
```diff
@@ -2,4 +2,4 @@ CreateModificationScript=Cr
 Console=Console
 EditScript=Editer le script
 ScriptName=Nom du script
-
+TabSize=Espace Tabulation
```
Comment (mathbagu) [inline gse-afs-ext-base/src/main/resources/lang/ModificationScript_fr.properties]: Are you sure of this translation?

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java [hunk 2]
```diff
@@ -104,6 +118,48 @@ public GroovyCodeEditor(Scene scene) {
...
+
+    private void onTabPressed() {
+        codeArea.deletePreviousChar();
+        int currentLine = codeArea.getCaretSelectionBind().getParagraphIndex();
+        int fromlineStartToCaret = codeArea.getText(currentLine, 0, currentLine, codeArea.getCaretColumn()).length();
+        codeArea.insertText(codeArea.getCaretPosition(), tabSpace(tabSpacesToComplete(fromlineStartToCaret)));
+    }
+
+    public void setTabSize(int size) {
+        codeArea.tabSize = size;
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: Assert that size is strictly positive

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java [hunk 3]
```diff
@@ -104,6 +118,48 @@ public GroovyCodeEditor(Scene scene) {
...
+            ke.consume();
+            String formatedClipBoardText = clipboard.getString().replaceAll(Character.toString('\t'), tabSpace(getTabSize()));
+            codeArea.insertText(codeArea.getCaretPosition(), formatedClipBoardText);
+        }
+    }
+
+    private void onTabPressed() {
+        codeArea.deletePreviousChar();
+        int currentLine = codeArea.getCaretSelectionBind().getParagraphIndex();
+        int fromlineStartToCaret = codeArea.getText(currentLine, 0, currentLine, codeArea.getCaretColumn()).length();
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: fromLine with an upper L

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java [hunk 4]
```diff
@@ -104,6 +118,48 @@ public GroovyCodeEditor(Scene scene) {
...
+
+    public void setTabSize(int size) {
+        codeArea.tabSize = size;
+    }
+
+    public int getTabSize() {
+        return codeArea.tabSize;
+    }
+
+    private String tabSpace(int size) {
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: Should be static

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java [hunk 5]
```diff
@@ -104,6 +118,48 @@ public GroovyCodeEditor(Scene scene) {
...
+        StringBuilder space = new StringBuilder();
+        for (int i = 0; i < size; i++) {
+            space.append(" ");
+        }
+        return space.toString();
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: Use StringUtils.repeat() ?

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java [hunk 6]
```diff
@@ -104,6 +118,48 @@ public GroovyCodeEditor(Scene scene) {
...
+    }
+
+    private int tabSpacesToComplete(int length) {
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: Should be static

This method seems complicated: what are you trying to do? To add the correct number of spaces from the current position to « complete » the tabulation?

int spacesToAdd = getTabSize() - (length % getTabSize());

Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java (hunk 5 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java (hunk 4 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java (hunk 3 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java (hunk 2 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: updated

File: gse-afs-ext-base/src/main/resources/lang/ModificationScript_fr.properties (hunk 1 above)
Comment (nashousni) [reply gse-afs-ext-base/src/main/resources/lang/ModificationScript_fr.properties]: updated

**Pull request merged**